File: src/main/java/com/EmosewaPixel/pixellib/materialSystem/lists/TextureTypes.java
package com.EmosewaPixel.pixellib.materialsystem.lists;

import com.EmosewaPixel.pixellib.materialsystem.types.TextureType;

import java.util.Collection;
import java.util.HashMap;

//This class contains functions for interacting with the global list of texture types
public final class TextureTypes {
    public static final HashMap<String, TextureType> textureTypes = new HashMap<>();

    public static void add(TextureType type) {
        textureTypes.put(type.toString(), type);
    }

    public static TextureType get(String name) {
        return textureTypes.get(name);
    }

    public static Collection<TextureType> getAll() {
        return textureTypes.values();
    }
}

File: src/main/java/com/emosewapixel/pixellib/materialsystem/lists/TextureTypes.kt
package com.emosewapixel.pixellib.materialsystem.lists

import com.emosewapixel.pixellib.materialsystem.types.TextureType

//This class contains functions for interacting with the global list of texture types
object TextureTypes {
    private val textureTypes = hashMapOf<String, TextureType>()

    @JvmStatic
    fun getAll() = textureTypes.values

    @JvmStatic
    fun add(type: TextureType) {
        textureTypes[type.toString()] = type
    }

    @JvmStatic
    operator fun get(name: String): TextureType? = textureTypes[name]
}